app/controllers/confirmation_controller.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+class ConfirmationController < ApplicationController
+  rescue_from ActiveRecord::RecordNotUnique, with: :error_not_unique
+
+  # new populates instance variables used by the new confirmation form.
+  def new
+    @term = Term.find(params[:term_id])
+    @categories = Category.all
+    @confirmation = Confirmation.new
+  end
+
+  # create receives the submission from the new confirmation form, creating the needed record with the help of various
+  # private methods.
+  def create
+    new_record = Confirmation.new(confirmation_params)
+
+    if new_record.save
+      # Now that the confirmation has saved, set the flag on the related Term record.
+      flag_term(params[:term_id]) if confirmation_flag?(params[:confirmation][:category_id])
+
+      # By this point, saving has succeeded
+      feedback_for(confirmation_flag?(params[:confirmation][:category_id]))
+      redirect_to terms_unconfirmed_path
+    else
+      error_cannot_save
+    end
+  end
+
+  private
+
+  def confirmation_params
+    params.require(:confirmation).permit(:term_id, :category_id, :user_id)
+  end
+
+  # feedback_for takes the result of the confirmation.save directive above and sets an appropriate flash message.
+  #
+  # The final else clause is likely to be difficult to provoke, so we are sending a Sentry message in that block in
```

Comment:
I think this line of documentation should be removed or moved as it does not reflect the current state of the method it is documenting.

Reply:
Good catch - I've rewritten this entire comment block. Thanks!